app/controllers/charges_controller.rb

```diff
@@ -0,0 +1,38 @@
+class ChargesController < ApplicationController
+  def create
```

Comment:
this controller action is getting pretty long - when you've got this many lines, it's a good indicator that we should probably abstract the logic elsewhere. You could put this inside a method on User or make it into a [Service Object](https://hackernoon.com/service-objects-in-ruby-on-rails-and-you-79ca8a1c946e)

The service object approach is probably the most common - it's a pattern we use regularly at work.

Reply:
okay ill work on addressing this in my final checkpoint